pkg/pillar/cmd/zedmanager/updatestatus.go

```diff
@@ -435,28 +441,31 @@ func doActivate(ctx *zedmanagerContext, uuidStr string,
 		if remaining < need {
 			var errStr string
 			var entities []*types.ErrorEntity
-			errSeverity := types.ErrorSeverityWarning
-			retryCondition := "Retry will be triggered when apps will shutdown: "
+			var domains []string
+			errSeverity := types.ErrorSeverityError
 			if remaining+halting < need {
 				errStr = fmt.Sprintf("Remaining memory bytes %d app instance needs %d",
 					remaining, need)
-				for _, st := range ctx.pubAppInstanceStatus.GetAll() {
-					status := st.(types.AppInstanceStatus)
-					entities = append(entities, &types.ErrorEntity{EntityID: status.UUIDandVersion.UUID.String(), EntityType: types.ErrorEntityAppInstance})
-					retryCondition = fmt.Sprintf("%s %s;", retryCondition, status.DisplayName)
-				}
 			} else {
-				errStr = fmt.Sprintf("App instance needs %d bytes but only have %d; waiting for halting app instances to free up %d bytes",
+				errStr = fmt.Sprintf("App instance needs %d bytes but only have %d; waiting for one or more halting app instances to free up %d bytes",
 					need, remaining, halting)
 				errSeverity = types.ErrorSeverityNotice
-				for _, st := range ctx.pubAppInstanceStatus.GetAll() {
-					status := st.(types.AppInstanceStatus)
-					if status.Activated || status.ActivateInprogress {
-						entities = append(entities, &types.ErrorEntity{EntityID: status.UUIDandVersion.UUID.String(), EntityType: types.ErrorEntityAppInstance})
-						retryCondition = fmt.Sprintf("%s %s;", retryCondition, status.DisplayName)
-					}
+			}
+			for _, st := range ctx.pubAppInstanceStatus.GetAll() {
+				status := st.(types.AppInstanceStatus)
+				if status.Activated || status.ActivateInprogress {
+					entities = append(entities, &types.ErrorEntity{EntityID: status.UUIDandVersion.UUID.String(), EntityType: types.ErrorEntityAppInstance})
+					domains = append(domains, status.DisplayName)
 				}
 			}
+			retryCondition := ""
+			if len(domains) > 0 {
+				if errSeverity == types.ErrorSeverityError {
+					//reduce error severity die to domains with status.Activated || status.ActivateInprogress
+					errSeverity = types.ErrorSeverityWarning
+				}
+				retryCondition = fmt.Sprintf("Retry will be triggered when one or more apps will shutdown: %s", strings.Join(domains, "; "))
```

Comment:
Please check with Sahil if he wants the list here or whether he will do that in the UI based on the list of entities you're setting.

Reply:
Checked with Sahil. UI will be adjusted to use ids in the upcoming sprint, so, it is reasonable to simplify the RetryCondition message.